xds/internal/test/xds_server_serving_mode_test.go

```diff
@@ -149,11 +116,19 @@ func (s) TestServerSideXDS_ServingModeChanges(t *testing.T) {
 	}()
 
 	// Wait for both listeners to move to "serving" mode.
-	if err := waitForModeChange(ctx, modeTracker, lis1.Addr(), xds.ServingModeServing); err != nil {
-		t.Fatal(err)
-	}
-	if err := waitForModeChange(ctx, modeTracker, lis2.Addr(), xds.ServingModeServing); err != nil {
-		t.Fatal(err)
+	for i := 0; i < 2; i++ {
```

Comment:
So _technically_ this will pass if one update channel gets a serving update twice.  That's probably impossible, so not the end of the world, but why not remove the loop and have two consecutive `select`s, one for each channel, instead?

Reply:
Done.